### PR TITLE
Track agda libraries using submodules, and permit parallelized builds

### PR DESCRIPTION
**Pull request by jwiegley**
If you run `nix-build --pure --cores 4` you will see that three files
currently fail to typecheck.

```
src/ModuleMorphism.agda:159,5-21
K !=< Setoid _s₁_307 _s₂_308
when checking that the expression (g ∘ f) (x ∙ᴵ y) has type
Setoid _s₁_307 _s₂_308
```

```
src/Cartesian/Homomorphisms.agda:57,40-49
Data.Product.Σ _A_36 _B_37 !=< Carrier (U (A ×ᴵ B))
when checking that the inferred type of an application
  Data.Product.Σ _A_36 _B_37
matches the expected type
  Carrier (U (A ×ᴵ B))
```

```
src/AlgebraicCats.agda:128,12-28
zero != c
when checking that the expression Z.leftSemimodule has type
LeftSemimodule R c ℓ
```

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[submodule "vendor/agda-stdlib"]
+	path = vendor/agda-stdlib
+	url = git@github.com:conal/agda-stdlib
+[submodule "vendor/agda-categories"]
```
Comment (conal) [inline .gitmodules:4]: What's "vendor" about?

**Comment by conal** [comment]
In a non-Nix setting, should ~/.agda/libraries be changed to local copies of these forks? My current ~/.agda/libraries:
```
~/git-repos/agda-stdlib/standard-library.agda-lib
~/git-repos/agda-categories/agda-categories.agda-lib
```

File: .gitmodules (hunk 1 above)
Comment (jwiegley) [reply .gitmodules:4]: This is just a typical Git naming convention when a project must include 3rd party source code in order to be able to build. For example, there is a utility out in the wild called `git-vendor` for helping to manage such things: https://github.com/brettlangdon/git-vendor.

But this subdirectory could be named anything you like.

**Comment by jwiegley** [comment]
In a non-Nix setting you'd change `~/.agda/libraries` to point to the copies inside the `vendor` directory, and then build and install from those directories I imagine.

**Comment by conal** [comment]
> In a non-Nix setting you'd change `~/.agda/libraries` to point to the copies inside the `vendor` directory, and then build and install from those directories I imagine.

Thanks. I've not used Git submodules. Should I edit/commit/push/sync in those local fork copies or in the local fork clones & have now?

**Comment by conal** [comment]
Is the name "vendor" a Nix convention?

**Comment by conal** [comment]
> If you run `nix-build --pure --cores 4` you will see that three files currently fail to typecheck.

Thanks. I moved `ModuleMorphisms` to `Old/`, commented out breaking code in `Cartesian.Homomorphisms`, and updated `AlgebraicCats`.

**Comment by conal** [comment]
```

bash-3.2$ dirs
~/Agda/cat-linear/src
bash-3.2$ cd ..
bash-3.2$ nix-build --pure --cores 4
error: getting attributes of path '/Users/conal/Agda/agda-stdlib': No such file or directory
(use '--show-trace' to show detailed location information)
bash-3.2$ nix-build --pure --cores 4 --show-trace
error: while evaluating the attribute 'buildInputs' of the derivation 'cat-linear-1.0' at /Users/conal/Agda/cat-linear/default.nix:16:3:
while evaluating the attribute 'buildCommand' of the derivation 'agdaWithPackages-2.6.1.1' at /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'text' of the derivation 'libraries' at /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating 'concatMapStringsSep' at /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/lib/strings.nix:88:5, called from /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/pkgs/build-support/agda/default.nix:14:10:
while evaluating anonymous function at /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/pkgs/build-support/agda/default.nix:14:36, called from undefined position:
while evaluating the attribute 'src' of the derivation 'standard-library-master' at /nix/store/ia8qfk4dyv21j956ccmaf6gy9zq7x650-source/pkgs/stdenv/generic/make-derivation.nix:192:11:
getting attributes of path '/Users/conal/Agda/agda-stdlib': No such file or directory
```

**Comment by jwiegley** [comment]
Is that using this PR branch?

**Comment by conal** [comment]
> Is that using this PR branch?

No, I guess it isn't.

**Comment by jwiegley** [comment]
@conal After updating this to the latest master, now just one failure:

```
src/AlgebraicCats.agda:132,12-28
zero != c
when checking that the expression Z.leftSemimodule has type
LeftSemimodule R c ℓ
```

**Comment by conal** [comment]
> @conal After updating this to the latest master, now just one failure:
> 
> ```
> src/AlgebraicCats.agda:132,12-28
> zero != c
> when checking that the expression Z.leftSemimodule has type
> LeftSemimodule R c ℓ
> ```

@jwiegley It looks like we're missing https://github.com/agda/agda-stdlib/pull/1323 .

**Comment by conal** [comment]
Oh! I bet the issue is that I haven't updated my agda-stdlib fork so that its master branch includes this 1323.

**Comment by conal** [comment]
Updated now, I think.

**Comment by conal** [comment]
```
bash-3.2$ pwd
/Users/conal/Agda/cat-linear
bash-3.2$ git pull
Already up-to-date.
bash-3.2$ nix-build --pure --cores 4 --show-trace
warning: unable to download 'https://cache.nixos.org/k47h5wfnwmdjrzynvq22j2x0n926bikb.narinfo': SSL peer certificate or SSH remote key was not OK (60); retrying in 279 ms
warning: unable to download 'https://cache.nixos.org/k47h5wfnwmdjrzynvq22j2x0n926bikb.narinfo': SSL peer certificate or SSH remote key was not OK (60); retrying in 625 ms
warning: unable to download 'https://cache.nixos.org/k47h5wfnwmdjrzynvq22j2x0n926bikb.narinfo': SSL peer certificate or SSH remote key was not OK (60); retrying in 1143 ms
warning: unable to download 'https://cache.nixos.org/k47h5wfnwmdjrzynvq22j2x0n926bikb.narinfo': SSL peer certificate or SSH remote key was not OK (60); retrying in 2433 ms
error: unable to download 'https://cache.nixos.org/k47h5wfnwmdjrzynvq22j2x0n926bikb.narinfo': SSL peer certificate or SSH remote key was not OK (60)
```

**Pull request merged**